### ISSUE TITLE
Remove Old Signs After Placing New Ones

### ISSUE THREAD

**Issue by jebaum** [neomake/neomake#320]
In `s:Make()`, it looks like all signs are removed before new ones are added. This causes the sign column to briefly flash off and on after Neomake finishes running, which is a bit annoying to the user. Would be nice if Neomake placed new signs before removing the old ones to prevent this.

**Comment by theamazingfedex** [comment]
Hmm, I agree that it's a bit jarring to have the list disappear, then reappear, but I think that maybe it would be even better if when the old gutter is removed, show a blank gutter but no errors, then once neomake completes, replace the blank gutter with the new gutter with the new signs, creating like a buffer

**Comment by blueyed** [comment]
Closing as dupe of #516.

@theamazingfedex 
> but I think that maybe it would be even better if when the old gutter is removed, show a blank gutter but no errors, then once neomake completes, replace the blank gutter with the new gutter with the new signs, creating like a buffer
 
That is the current behavior isn't it?

Will have to investigate, but I thought it would clear signs when the job starts already?!

**Comment by cjlucas** [comment]
The signs do clear in the gutter when the job starts, but I think the main issue @jebaum was getting at is the jarring effect caused by the gutter being removed and redrawn when jobs are run.

**Comment by blueyed** [comment]
So it should not replace existing signs, but keep them in case they would appear at the same place again?

**Comment by cjlucas** [comment]
The issue is that the gutter is removed when the job begins and is immediately redrawn (if there are still errors). This causes the window to shift, which is quite jarring. If it's possible to keep the gutter open until the job completes and removing it only if there are no errors/warnings, I think that would fix the issue.

In addition, it would be nice if the quickfix buffer acted the same way. In that it would only disappear once a job finishes successfully.

**Comment by cjlucas** [comment]
So I looked into this issue a little bit more. Being unfamiliar with vim's development tools, I didn't know about the `sign` command, and how neomake uses it. Apparently this was an issue in mainline vim that was resolved recently (see: https://github.com/vim/vim/issues/117).

I'll file an issue upstream to see if this can be added to neovim as well, but I don't believe anything needs to be done on your end @blueyed.
